backend/accounts/baseaccount.go

```diff
@@ -133,6 +137,24 @@ func (account *BaseAccount) Notes() *notes.Notes {
 	return account.notes
 }
 
+// ProposeTxNote implements accounts.Account.
+func (account *BaseAccount) ProposeTxNote(note string) {
+	account.proposedTxNoteMu.Lock()
+	defer account.proposedTxNoteMu.Unlock()
+
+	account.proposedTxNote = note
+}
+
+// GetAndClearProposedTxNote returns the note previously set using ProposeTxNote(). If none was set,
+// the empty string is returned. The proposed note is cleared by calling this function.
+func (account *BaseAccount) GetAndClearProposedTxNote() string {
+	account.proposedTxNoteMu.Lock()
+	defer account.proposedTxNoteMu.Unlock()
+	note := account.proposedTxNote
+	account.proposedTxNote = ""
+	return note
```

Comment:
This would also have worked:

```go
account.proposedTxNoteMu.Lock()
defer func() {
  account.propostedTxNote = ""
  account.proposedTxNoteMu.Unlock()
}()
return account.proposedTxNote
```

Here's an example: https://play.golang.org/p/D1gxxJ0PleD

Reply:
nice